pyqtorch/hamiltonians/evolution.py

```diff
@@ -372,3 +373,27 @@ def tensor(
             return evolved_op
         else:
             return expand_operator(evolved_op, self.qubit_support, full_support)
+
+    def jacobian(
+        self, values: dict[str, Tensor] = dict(), embedding: Embedding | None = None
+    ) -> Tensor:
+        """
+        Get the corresponding unitary of the jacobian.
+
+        Arguments:
+            values: Parameter value.
+
+        Returns:
+            The unitary representation of the jacobian.
+        """
+
+        hamiltonian: torch.Tensor = self.create_hamiltonian(values, embedding)  # type: ignore [call-arg]
+        time_evolution: torch.Tensor = (
+            values[self.time] if isinstance(self.time, str) else self.time
+        )  # If `self.time` is a string / hence, a Parameter,
+        # we expect the user to pass it in the `values` dict
```

Comment:
Maybe it's worth checking the `self.time` type (although it's annotated in the class `__init__` method that it can be `str` or `Tensor`, nothing prevents it not being. So either enforcing that on instantiation or here would be good to spot early errors.